packages/nextcloud/lib/src/webdav/client.dart

```diff
@@ -62,27 +28,17 @@ class WebDavClient {
   String? _token;
 
   Future<http.StreamedResponse> _send(
-    String method,
-    Uri url, {
-    Stream<List<int>>? dataStream,
-    Uint8List? data,
-    Map<String, String>? headers,
-  }) async {
-    final request = WebDavRequest(
-      method,
-      url,
-      data: data,
-      dataStream: dataStream,
-      headers: headers,
-    );
-
+    http.BaseRequest request,
+  ) async {
     final authentication = rootClient.authentications?.firstOrNull;
     if (authentication != null) {
       request.headers.addAll(
         authentication.headers,
       );
     }
 
+    request.headers[HttpHeaders.contentTypeHeader] = 'application/xml';
```

Comment:
Shouldn't the header be set in the requests directly? Otherwise we have to set it again in the RequestManager.

Reply:
Yes; in an ideal world the contentTypeHeader and authentication headers would be set for every request and if you want I can also do it in this PR.

Sadly this will not help us in any way with the RequestManager. The CSRF bug still requires some async computation before the request is finalized. 
I'm currently planning to allow the request manager `wrap` method to take a HttpClient as a parameter. This way the `wrapWebDav` method can provide a client that adds the CSRF workaround (probably the `WebDavClient` once it also extends HttpBaseClient). All other wrap implementations would provide the base nextcloud client form the account as it is already done.

This also allows us to further decouple the RequestManager from the neon_framework by reducing the reliance on the `Account` model allowing us to separate it into another package when needed.

Reply:
Should I change the headers now or just leave them in the client as it will be needed anyways to set the CSRF token?

Reply:
Since we will no longer need the CSRF hack at some point the whole async thing will go away, so I'd prefer if the headers are already set in the request because we will not need the custom client anymore.